fabric-testmods/java/net/fabricmc/fabric/enchantments/EnchantmentMod.java

```diff
@@ -0,0 +1,55 @@
+package net.fabricmc.fabric;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.enchantment.enchantment.EnchantingPowerProvider;
+import net.fabricmc.fabric.api.enchantment.enchantment.FabricEnchantment;
+import net.fabricmc.fabric.api.enchantment.enchantment.FabricEnchantmentTarget;
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.BlockState;
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.enchantment.EnchantmentTarget;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.World;
+
+public class EnchantmentMod implements ModInitializer {
+	@Override
+	public void onInitialize() {
+		Registry.register(Registry.ENCHANTMENT, new Identifier("fabric", "test"), new TestEnchantment(Enchantment.Weight.COMMON, EquipmentSlot.values()));
+		Registry.register(Registry.BLOCK, new Identifier("fabric", "enchanting_power_block"), new TestBlock(Block.Settings.copy(Blocks.STONE)));
+	}
+
+	static class TestEnchantment extends Enchantment implements FabricEnchantment {
+		protected TestEnchantment(Weight enchantment$Weight_1, EquipmentSlot[] equipmentSlots_1) {
+			super(enchantment$Weight_1, EnchantmentTarget.ALL, equipmentSlots_1);
+		}
+
+		@Override
+		public FabricEnchantmentTarget getEnchantmentTarget() {
+			return new TestTarget();
```

Comment:
Why create a new instance everytime? The test mod will be used as an example, and this seems unnecessary to me.

Reply:
ok I'm going to review the test stuff again ;)